networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/topics/validation/AttestationValidator.java

```diff
@@ -211,6 +229,23 @@ private UnsignedLong maximumBroadcastTimeMillis(final UnsignedLong attestationSl
     return secondsToMillis(lastAllowedTime).plus(MAXIMUM_GOSSIP_CLOCK_DISPARITY);
   }
 
+  StateProvider createStateProvider() {
+    return createStateProvider(recentChainData);
+  }
+
+  @VisibleForTesting
+  static StateProvider createStateProvider(final RecentChainData recentChainData) {
+    LoadingCache<Bytes32, SafeFuture<Optional<BeaconState>>> cache =
+        CacheBuilder.newBuilder().build(CacheLoader.from(recentChainData::retrieveBlockState));
```

Comment:
Shouldn't `Store` automatically take care of caching for us?  We're also only loading one state and creating a new cache each time so I don't think this can wind up ever being useful.

Reply:
Oh I see we do reuse it in `SignedAggregateAndProofValidator`.  I suspect leaving the caching to the `Store` is likely to be the right answer still.

Reply:
Simplified this. It felt weird to me to potentially regenerate the same state twice, but in reality the `Store` should still have it cached.